Core & Internals: Add support to inequality filters in metadata queries Fix #3750

gabrielefronze opened the pull request:
Work in progress PR about the integration of the Rucio Inequality Engine (https://github.com/gabrielefronze/rucio-inequality-engine) within `rucio/rucio`.

gabrielefronze committed:
Create file

gabrielefronze committed:
First commit

gabrielefronze committed:
Using specific imports

gabrielefronze committed:
Fix imports

bziemons reviewed (commented):
In reply to https://github.com/rucio/rucio/pull/4075/commits/0c9bcbfb72b1de58d8d1bbef69089945224c55b8:

FYI: you don't need this when you use the `tools/add_header` script and it adds the `# -*- coding: utf-8 -*-` to the file header.

gabrielefronze commented:
@bziemons I wasn't sure about adding the custom encoding, thanks for letting me know!

gabrielefronze committed:
Fixing syntax issues

Fix syntax

Fixing syntax flaws

Removing type hints

Fix syntax

Removing return hints

Remove remaining type hint

Fix sysntax

gabrielefronze committed:
Removing unused import

gabrielefronze committed:
Fixing DataIndentifier resolution

Removing typo

gabrielefronze committed:
Adding stream session

gabrielefronze committed:
WIP test

gabrielefronze committed:
WIP 2

gabrielefronze committed:
WIP 3

gabrielefronze committed:
Moving to read session

gabrielefronze committed:
WIP 4

gabrielefronze committed:
WIP 5

gabrielefronze committed:
Converting list to args

gabrielefronze committed:
Converting dict to args

gabrielefronze committed:
Trying filter_by

gabrielefronze committed:
Tring sqlalchemy filters from text

gabrielefronze committed:
Adding necessary import

gabrielefronze committed:
Revert "Removing metadata expansion"

This reverts commit a58a79245c034fac661d7ec2ccf25c0a1d40446c.

gabrielefronze committed:
Removing metadata expansion

gabrielefronze committed:
WIP 6

gabrielefronze committed:
Switching to tablename
